pkg/aggregator/aggregator.go

```diff
@@ -479,9 +479,11 @@ func (agg *BufferedAggregator) run() {
 			agg.flush()
 			addFlushTime("MainFlushTime", int64(time.Since(start)))
 			aggregatorNumberOfFlush.Add(1)
-		case sample := <-agg.dogstatsdIn:
-			aggregatorDogstatsdMetricSample.Add(1)
-			agg.addSample(sample, timeNowNano())
+		case samples := <-agg.dogstatsdIn:
+			for _, sample := range samples {
+				aggregatorDogstatsdMetricSample.Add(1)
+				agg.addSample(sample, timeNowNano())
```

Comment:
Because these were getting buffered, the timestamp is now a little less precise in its actual "meaning", as it no longer tells us much about how samples were added to the aggregator over time, as they're added in batches. Wondering if we should do:
```go
ts := timeNowNano() 
for _, sample := range samples {
				aggregatorDogstatsdMetricSample.Add(1)
				agg.addSample(sample, ts)
}
``` 
to avoid making additional, maybe mostly useless, calls... Not sure, just a thought given the optimizing nature of this PR.

Also, unrelated to your work, the aggregator samplers don't look thread safe. I'd have to take a closer look to see if that may be problematic.